app/services/openlibrary_service.rb

```diff
@@ -0,0 +1,34 @@
+require 'set'
+
+class OpenlibraryService
+  include HTTParty
+  base_uri ENV['OPEN_LIBRARY_URI']
+
+  default_params format: :json, jscmd: 'data'
+
+  def search_by_isbn(isbn)
+    isbn_string = "ISBN:#{isbn}"
+    options = { query: { bibkeys: isbn_string } }
+    response = self.class.get('/books', options)
+
+    return parse_response_book(response[isbn_string], isbn) if response.success?
+
+    raise CustomErrors::BookNotFoundError, response.response
+  end
+
+  private
+
+  def parse_response_book(book, isbn)
+    raise CustomErrors::BookNotFoundError, 'Invalid response book' unless
+      book&.include_all?('title', 'subtitle', 'number_of_pages', 'authors') &&
```

Comment:
When evaluate conditions, pay attention to `logical weight`. 
Ruby is lazy, and in this condition `A && B` if (A is false), then B will never be evaluated.
`logical weight` says: if you have 2 conditions, use the most "descartable" option first.
In this case, it's more easy to ask `book['authors'].is_a?(Array)` than for another condition.